src/node_watchdog.h

```diff
@@ -54,24 +61,54 @@ class Watchdog {
   bool* timed_out_;
 };
 
-class SigintWatchdog {
+class SigintWatchdogBase {
+ public:
+  virtual ~SigintWatchdogBase() = default;
+  virtual SignalPropagation HandleSigint() = 0;
+};
+
+class SigintWatchdog : public SigintWatchdogBase {
  public:
   explicit SigintWatchdog(v8::Isolate* isolate,
                           bool* received_signal = nullptr);
   ~SigintWatchdog();
   v8::Isolate* isolate() { return isolate_; }
-  void HandleSigint();
+  SignalPropagation HandleSigint() override;
 
  private:
   v8::Isolate* isolate_;
   bool* received_signal_;
 };
 
+class TraceSigintWatchdog : public HandleWrap, public SigintWatchdogBase {
+ public:
+  static void Init(Environment* env, Local<v8::Object> target);
+  static void New(const v8::FunctionCallbackInfo<v8::Value>& args);
+  static void Start(const v8::FunctionCallbackInfo<Value>& args);
+  static void Stop(const v8::FunctionCallbackInfo<Value>& args);
+
+  SignalPropagation HandleSigint() override;
+
+  SET_NO_MEMORY_INFO()
```

Comment:
Can you implement the `MemoryInfo` method of this class since it has `uv_async_t`? You can check out `src/memory_tracker.h` on how to use it.

(on a side note, we probably want to write more stuff in the variant of `TrackField` that takes a `uv_async_t` so that the `UV_HANDLE_FIELDS` show up in heap snapshots for debugging)

Reply:
I'm confused that `SET_SELF_SIZE` might already include the inlined uv_async_t handle, is it still required to track the field? IIUC the additional space shall be tracked manually if we're referencing it with a pointer, like `uv_async_t*`.

Reply:
When you track a non-pointer field its size would be subtracted from the size of the parent node and a new node will be created and displayed in the snapshot. Therefore you can see its value in the heap snapshot (this becomes more useful when the handle is deeply tracked, then you can discover other handles that points to similar handles or data or group them)

Reply:
Updated, please take a look again :D.